spark/src/main/scala/org/apache/spark/sql/delta/files/DelayedCommitProtocol.scala

```diff
@@ -67,7 +70,7 @@ class DelayedCommitProtocol(
   // since there's no guarantee the stats will exist.
   @transient val addedStatuses = new ArrayBuffer[AddFile]
 
-  val timestampPartitionPattern = "yyyy-MM-dd HH:mm:ss[.S]"
+  val timestampPartitionPattern = "yyyy-MM-dd HH:mm:ss[.SSSSSS][.S]"
```

Comment:
This also addresses an issue of microsecond partitions; not that I'd reccomend identity partitions on microseconds due to extremely small files but it is technically supported by the delta protocol.